Buf_write: Fix BE.uint48 and LE.uint48.

adatario opened the pull request:
Arguments to logical shift are provided in bits, not bytes.

No tests added yet. It would be nice to test the `Buf_write` <-> `Buf_read` roundtrip. I'd be happy to add them, but am not sure where the best place would be.

talex5 commented:
`tests/buf_write.md` would be the best place for some tests. There are already tests for the reading side:

https://github.com/ocaml-multicore/eio/blob/80f5352526b260e9c32d7110844c7e7408e905f4/tests/buf_reader.md?plain=1#L418-L419

Note: this code is from [Faraday](https://github.com/inhabitedtype/faraday/blob/c66fc020fdc2327556c36fc1e912524b427d7da8/lib/faraday.ml#L290-L297), so we should probably report it there too.

Thanks!

talex5 commented:
I added some round-trip tests, and also fixed a second bug in `BE.uint48` :scream:

I also added `Eio.Buf_read.uint8`, which was missing.

adatario commented:
> I added some round-trip tests

Cool, thanks!

> and also fixed a second bug in BE.uint48 :scream:

:) Interesting that this was never encountered in Faraday.
